src/mpi/comm/commutil.c

```diff
@@ -43,6 +43,52 @@ struct MPIR_Comm_hint_fn_elt {
 };
 static struct MPIR_Comm_hint_fn_elt *MPID_hint_fns = NULL;
 
+int MPIR_Comm_init(void)
```

Comment:
The name `MPIR_Comm_init` is confusing because this is only for the builtin comms. I think this code should be under `init` directory, not in `commutil.c` because it is not general comm code.

Reply:
I agree that `MPIR_Comm_init` is not good as it is too similar to `MPIR_Comm_create` and easily to be confused with runtime comm init. However, I would like the code inside `src/mpi/comm/` sub tree, as all its structure field logic is defined and accessed there.

I am putting some of the init functions inside `src/mpi/init` now simply as a transition measure. Ideally, the `init` folder should only concern about the component init orders, the actual init details should be grouped with their other code in the same component -- a goal of component abstraction.

Reply:
@yfguo How about `MPIR_comm_init_local` or (as I now leaning toward) `MPIR_comm_init_stage1`?

Reply:
We can make a separate c unit, e.g. `comm_init.c` to host init and finalize code for the communicator system.

Reply:
`MPIR_Init_builtin_comms`